functions/supplier/zeromq-supplier/src/main/java/org/springframework/cloud/fn/supplier/zeromq/ZeroMqSupplierProperties.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2016-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.fn.supplier.zeromq;
+
+import java.util.Arrays;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+import org.zeromq.SocketType;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.validation.annotation.Validated;
+
+@ConfigurationProperties("zeromq.supplier")
+@Validated
+public class ZeroMqSupplierProperties {
+
+	private String connectUrl;
```

Comment:
I wonder what is wrong with the `bindPort` property as well...
Don't you think that someone may consider to have source app as a server for ZeroMq socket?

Reply:
Consider to expose also:
```
	/**
	 * Specify a {@link Duration} to delay consumption when no data received.
	 * @param consumeDelay the {@link Duration} to delay consumption when empty;
	 *                     defaults to {@link #DEFAULT_CONSUME_DELAY}.
	 */
	public void setConsumeDelay(Duration consumeDelay) {
```

Reply:
@artembilan  I was expecting this supplier to only connect to a server and receive messages from it. Should it also be able to configure a server? I guess I was expecting the consumer equivalent (forthcoming) to do that.

Reply:
Will add the `consumeDelay` as a configurable option

Reply:
No, `consumer` is just a publisher and to not let it to lose messages it is better to not have it responsible for a void state on the server side. At least that's my feeling about ZeroMq socket since we don't have a persistence in between. 
My point is here that with ZeroMq supplier as a server you kind introduce a middleware between ZeroMq publishers and persistent binder. So, if no server backed by our supplier, no middleware to produce into the binder!